pkg/operator/sync.go

```diff
@@ -1498,16 +1498,16 @@ func (optr *Operator) getImageRegistryPullSecrets() ([]byte, error) {
 	// Check if image registry exists, if it doesn't we no-op
 	co, err := optr.configClient.ConfigV1().ClusterOperators().Get(context.TODO(), "image-registry", metav1.GetOptions{})
 
+	// returning no error in certain cases because image registry may become optional in the future
+	// More info at: https://issues.redhat.com/browse/IR-351
+	if co == nil || apierrors.IsNotFound(err) {
```

Comment:
is there a scenario where co == nil but apierrors.IsNotFound == false that we still want to error out? or is the only scenario where co == nil when the image-registry is not found?

Reply:
Hmm good point. I am unsure if `co == nil` and `apierrors.IsNotFound == false` will never not coexist.  Originally, I was using the operator object returned by the `Get()` call as a way to see if the image registry is ready(and existing) - which was a bad assumption, as the `Get()` call can return an error if the registry isn't present. Perhaps, I should just move the `co==nil` to the degrade carveout where I'm catching the other errors(or just remove it altogether and trust the error object completely). Thoughts?